extension-manifest-v3/src/background/session.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+import { store } from 'hybrids';
+
+import Options from '/store/options.js';
+import { UPDATE_SESSION_ACTION_NAME } from '/store/session.js';
+
+import { session, COOKIE_DOMAIN } from '/utils/api.js';
+
+// Trigger options sync every hour
+const ALARM_SYNC_OPTIONS = 'session:sync:options';
+const ALARM_SYNC_OPTIONS_RATE = 1 * 60; // 1 hour in minutes
+
+// Try to check cookies every 30 days if the user is still logged in
+const ALARM_UPDATE_SESSION = 'session:update';
+const ALARM_UPDATE_SESSION_DELAY = 1000 * 60 * 24 * 30; // 30 days in milliseconds
+
+// Observe cookie changes (login/logout actions)
+chrome.cookies.onChanged.addListener(async ({ cookie, removed }) => {
```

Comment:
I've heard that this listener is not very reliable - need to be tested in all browser. Most importantly Firefox and Safari.

Reply:
I'll check that.